### PR TITLE
Feature: add health check to stellar observer

### PR DESCRIPTION
**Pull request by marcelosalloum**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Add a `/health` endpoint to the Stellar Observer deployment.

### Why

To enable people to check the status of the Stellar Observer.

### CODE REVIEW

File: service-runner/src/main/java/org/stellar/anchor/platform/ServiceRunner.java [hunk 1]
```diff
@@ -60,7 +61,17 @@ static ConfigurableApplicationContext startSepServer() {
   }
 
   static void startStellarObserver() {
-    StellarObservingService.start();
+    String strPort = System.getProperty("OBSERVER_HEALTH_SERVER_PORT");
+    int port = DEFAULT_OBSERVER_HEALTH_SERVER_PORT;
+    if (strPort != null) {
+      port = Integer.parseInt(strPort);
+    }
+    String contextPath = System.getProperty("SEP_CONTEXTPATH");
```
[the remark this answers is not in the document]
Comment (lijamie98) [reply service-runner/src/main/java/org/stellar/anchor/platform/ServiceRunner.java:69]: 💡 We can add the context path configuration in version 2.x

File: service-runner/src/main/java/org/stellar/anchor/platform/ServiceRunner.java [hunk 2]
```diff
@@ -60,7 +61,17 @@ static ConfigurableApplicationContext startSepServer() {
...
+    if (contextPath == null) {
+      contextPath = DEFAULT_CONTEXTPATH;
```
Comment (lijamie98) [inline service-runner/src/main/java/org/stellar/anchor/platform/ServiceRunner.java:71]: 💡 `STELLAR_OBSERVER_DEFAULT_CONTEXTPATH`?

File: service-runner/src/main/java/org/stellar/anchor/platform/ServiceRunner.java [hunk 3]
```diff
@@ -7,6 +7,7 @@
 public class ServiceRunner {
   public static final int DEFAULT_SEP_SERVER_PORT = 8080;
   public static final int DEFAULT_ANCHOR_REFERENCE_SERVER_PORT = 8081;
+  public static final int DEFAULT_OBSERVER_HEALTH_SERVER_PORT = 8083;
```
[the remark this answers is not in the document]
Comment (marcelosalloum) [reply service-runner/src/main/java/org/stellar/anchor/platform/ServiceRunner.java:10]: 8082 is already taken by the metrics.

File: platform/src/main/java/org/stellar/anchor/platform/controller/CirclePaymentObserverController.java [hunk 4]
```diff
@@ -18,6 +19,7 @@
 
 @RestController
 @RequestMapping("/circle-observer")
+@Profile("default")
```
[the remark this answers is not in the document]
Comment (marcelosalloum) [reply platform/src/main/java/org/stellar/anchor/platform/controller/CirclePaymentObserverController.java:22]: Yeah, we have the `default` profile, that's used by the SEP service when no profile is configured.

File: service-runner/src/main/java/org/stellar/anchor/platform/ServiceRunner.java (hunk 2 above)
Comment (marcelosalloum) [reply service-runner/src/main/java/org/stellar/anchor/platform/ServiceRunner.java]: Addressed in https://github.com/stellar/java-stellar-anchor-sdk/pull/602/commits/1a530f5a0e2688076187295c551b7d59d4458d5a.

File: service-runner/src/main/java/org/stellar/anchor/platform/ServiceRunner.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (marcelosalloum) [reply service-runner/src/main/java/org/stellar/anchor/platform/ServiceRunner.java]: Updated in https://github.com/stellar/java-stellar-anchor-sdk/pull/602/commits/1a530f5a0e2688076187295c551b7d59d4458d5a.

File: service-runner/src/main/java/org/stellar/anchor/platform/ServiceRunner.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (lijamie98) [reply service-runner/src/main/java/org/stellar/anchor/platform/ServiceRunner.java]: Ah... I see.

**Review by lijamie98 — APPROVED** [review]
LGTM

**Pull request merged**